tests/memealyzer.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { test, expect, WebSocket } from "@playwright/test";
+
+let uri = process.env.WEBAPP_ENDPOINT || "http://localhost:1080/";
+
+interface Image {
+  id: string;
+  Id: string;
+}
+
+interface ImageEvent {
+  arguments: Image[];
+}
+
+test("Add Meme Test", async ({ page }) => {
+
+  // This test will add a meme and check SignalR response
+
+  await page.goto(uri);
+
+  let webSocket: WebSocket;
+
+  page.on("websocket", (ws) => {
+    webSocket = ws;
+  });
+
+  const add = page.locator(".form-row button").first();
+
+  await page.waitForSelector(".spinner-grow", { state: "detached" });
+
+  const cards = page.locator(".card");
+
+  const beforeCardCount = await cards.count();
+
+  let expectedId: string;
+  let actualId: string;
+
+  await Promise.all([
+    page
+      .waitForResponse(
+        (response) =>
+          response.url().endsWith("/image") && response.status() === 200
+      )
+      .then(async (response) => {
+        const image = await response.json() as Image;
+        expectedId = image.id;
+      }),
+    await add.click(),
```

Comment:
this `await` makes the `Promise.all` useless because it tries to resolve the click call, which might results in flaky tests. I would recommend to remove the await there in line 47.